compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -571,6 +571,32 @@ impl<'value, 'interner> Display for ValuePrinter<'value, 'interner> {
                         Token::QuotedType(id) => {
                             write!(f, " {}", self.interner.get_quoted_type(*id))?;
                         }
+                        Token::InternedExpr(id) => {
+                            let expr = self.interner.get_expression_kind(*id);
+                            let expr =
+                                remove_interned_in_expression_kind(self.interner, expr.clone());
+                            write!(f, " {}", expr)?;
+                        }
+                        Token::InternedStatement(id) => {
+                            let stmt = self.interner.get_statement_kind(*id);
+                            let stmt =
+                                remove_interned_in_statement_kind(self.interner, stmt.clone());
+                            write!(f, " {}", stmt)?;
+                        }
+                        Token::InternedLValue(id) => {
+                            let lvalue = self.interner.get_lvalue(*id, Span::default());
+                            let lvalue = remove_interned_in_lvalue(self.interner, lvalue.clone());
+                            write!(f, " {}", lvalue)?;
+                        }
+                        Token::InternedUnresolvedTypeData(id) => {
+                            let typ = self.interner.get_unresolved_type_data(*id);
+                            write!(f, " {}", typ)?;
+                        }
+                        Token::UnquoteMarker(id) => {
+                            let hir_expr = self.interner.expression(id);
+                            let expr = hir_expr.to_display_ast(self.interner, Span::default());
+                            write!(f, " {}", expr.kind)?;
+                        }
```

Comment:
Instead of handling these cases here we should create `Value` variants like `Value::TypedExpr(id)` for this case then recursively call `fmt` on those. Currently we'd print a quoted expr containing an unquoted typed expr well, but the base case for typed exprs below on line 661 is unchanged so normal TypedExpr values would be printed differently.

If we create values for these cases and format those recursively instead this inner match can hopefully be a bit smaller and we can share more code.

Reply:
Great idea, I just pushed a commit for it.